File: src/main/node/colourdisplay.js
'use strict'

const Colour = require('./colour.js');
const Display = require('./display.js');
const { ESC, CLR, RST, REVON, REVOFF, BLINK } = require('./ansi-term.js');

const DEFAULT = ESC+Colour.WHITE+'m'+ESC+Colour.BLACK_BG+'m';

module.exports =

class ColourDisplay extends Display {
  constructor(game) {
    super(game);
  }
  
  refresh(output, colour) {
    this.write(CLR);
    this.write(DEFAULT);
    for (let i=0;i<output.length;i++) {
      if (colour[i] >0) { 
        this.write(ESC+parseInt(colour[i])+'m');
      }
      this.write(output[i]);
      this.write(RST);
      this.write(DEFAULT);  
    }
    this.write(RST);
    this.write("\n");
  }
}

File: src/main/python/colourdisplay.py
from display import Display
from colour import Colour


class ColourDisplay(Display):
    DEFAULT = Display.ESC + str(Colour.WHITE) + "m"
    DEFAULT += Display.ESC + str(Colour.BLACK_BG) + "m"

    def refresh(self, video_output, colour_out=None):
        self.output = ""
        self.write_stream(Display.CLR)
        self.write_stream(ColourDisplay.DEFAULT)
        for index, video_data in enumerate(video_output):
            if colour_out and colour_out[index] > 0:
                self.write_stream(Display.ESC + str(colour_out[index]) + "m")
                self.write_stream(video_data)
                self.write_stream(Display.RST)
                self.write_stream(ColourDisplay.DEFAULT)
            else:
                self.write_stream(video_data)
        self.write_stream(Display.RST)
        self.write_stream("\n")
